desc/objectives/getters.py

```diff
@@ -213,12 +215,8 @@ def _is_any_instance(things, cls):
         return any([isinstance(t, cls) for t in things])
 
     # Equilibrium
-    if (
-        hasattr(thing, "Ra_n")
-        and hasattr(thing, "Za_n")
-        and hasattr(thing, "Rb_lmn")
-        and hasattr(thing, "Zb_lmn")
-        and hasattr(thing, "L_lmn")
+    if {"Rb_lmn", "Zb_lmn", "L_lmn", "Ra_n", "Za_n"} <= set(
```

Comment:
would it make sense to do a local import for proper instance checking here?

Reply:
That might make the code more clear, although it won't reduce the number of lines of code or anything. And technically these linear constraints only care that the fixed attribute exists (which we are checking for), not that the thing is a particular class. So I could go either way

Reply:
I would vote for a local import and proper instance check, or to break this up like 
```python
if "Rb_lmn" in params and "R_lmn" in params:
    constraints += (FixBoundaryR(eq),)
if "L_lmn" in params:
    constraints += (FixLambdaGauge(eq),)
...
```
so that each constraint is added only if all of the params it requires are there